pkg/cluster/cluster.go

```diff
@@ -295,6 +295,8 @@ func (c *Cluster) run(stopC <-chan struct{}) {
 				break
 			}
 
+			garbagecollection.CollectDeadMemberServices(c.cluster.Name, c.cluster.Namespace, c.config.KubeCli, c.members, c.logger)
```

Comment:
> We can't have another routine changing the membership while gc reading it. 

This is a typical GC issue. You just need to stop the world while doing GC. But we should not spread out the GC code into our codebase just to make GC "concurrently". The benefit is minimal for us, and it makes codebase harder to maintain.